include/deal.II/base/derivative_form.h

```diff
@@ -319,6 +325,20 @@ DerivativeForm<order,dim,spacedim,Number>::times_T_t (const Tensor<2,dim,Number>
 }
 
 
+
+template <int order, int dim, int spacedim, typename Number>
+inline
+double
+DerivativeForm<order,dim,spacedim,Number>::norm () const
+{
+  double sum_of_squares = 0;
```

Comment:
Same here, I would expect `real_type` since the underlying number can be an auto differentiation type, can't it?